bats/tests/containers/published-ports.bats

```diff
@@ -0,0 +1,45 @@
+load '../helpers/load'
+
+skip_unless_host_ip() {
+    if using_windows_exe; then
+        HOST_IP=$(netsh.exe interface ip show addresses 'vEthernet (WSL)' | grep -Po 'IP Address:\s+\K[\d.]+')
+    else
+        # TODO determine if the Lima VM has its own IP address
+        HOST_IP=""
+    fi
+    if [[ -z $HOST_IP ]]; then
+        skip "Test requires a routable host ip address"
+    fi
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+run_container_with_published_port() {
+    local container_image="nginx"
+    ctrctl pull "$container_image"
+    ctrctl run -d -p "$@" --restart=no "$container_image"
```

Comment:
Please use `IMAGE_NGINX` so we can optionally pull from `ghcr.io` instead of `docker.io`:

```suggestion
    ctrctl pull "$IMAGE_NGINX"
    ctrctl run -d -p "$@" --restart=no "$IMAGE_NGINX"
```